pkg/imgpkg/cmd/images_lock_writer.go

```diff
@@ -0,0 +1,46 @@
+package cmd
+
+import (
+	"path/filepath"
+
+	"github.com/cppforlife/go-cli-ui/ui"
+	ctlbundle "github.com/k14s/imgpkg/pkg/imgpkg/bundle"
+	"github.com/k14s/imgpkg/pkg/imgpkg/lockconfig"
+)
+
+//go:generate go run github.com/maxbrunsfeld/counterfeiter/v6 github.com/cppforlife/go-cli-ui/ui.UI
+
+func NewImagesLockWriter(ui ui.UI) *ImagesLockWriter {
+	return &ImagesLockWriter{ui: ui}
+}
+
+type ImagesLockWriter struct {
```

Comment:
Instead of having a do-er object (i.e. doesnt really represent a real noun), it would be good to consider what if this object was ImagesLock (a bundle specific one, since it's in bundle package). you could have something like this:

```
NewImagesLock(*lockconfig.ImagesLock, imgRetriever, relativeToRepo) bundle.ImagesLock
ImagesLock#WriteToPath(path, ui)
ImagesLock#localize(*lockconfig.ImagesLock) (bool, error)
```

this way we are encapsulating behaviour of what it means to be a bundle's imagelock in one class. this also avoid backward ref to the bundle.

Reply:
(i dont think we need to depedency inject it into Bundle but if we really want to we can create tiny factory/function that knows how to spin one up).